Add support for Import Statements

matthias-mar opened the pull request.

ichxorya commented:
Very cool feature! Hopefully this can be reviewed and merged into the code base.

matthias-mar commented:
**Changes:**
- Added dedicated file path during compilation, since imports rely on well-formed file paths to locate imported specifications
- Introduced model qualifier concept:
    - Users can import elements using their model name as a qualifier (import modelName#elementName from "modelName.use"), to avoid name conflicts with duplicate element identifiers
    - Uniqueness of identifiers is only guaranteed between a model and its direct imports, transitively imported elements must be accessed using the qualified name syntax (model A imports class B from model C, class B has attribute x with imported data type D from model E -> in order to assign a value to attribute x, the constructor must be called like !set b.x := E#D(1,1)
- Added Shell integration tests for models with imports

h-man2 reviewed (changes requested):
See comments

Please add dev documentation about the internals

ichxorya commented on use-core/src/main/java/org/tzi/use/parser/ocl/ASTSimpleType.java:
```diff
@@ -38,6 +38,10 @@ public ASTSimpleType(Token name) {
         fName = name;
     }
 
+    public Token getfName() {
```
I saw many `f` in the code base and still doesn't get the convention, do you know why there were `f` variables?

matthias-mar commented on use-core/src/main/java/org/tzi/use/parser/ocl/ASTOperationExpression.java:
```diff
@@ -596,7 +611,8 @@ private Expression genObjOperation(Context ctx, MClassifier srcClassifier, Expre
 
         // check for constructor call
-        MClassifier cf = ctx.model().getClassifier(opname);
+        MClassifier cf = srcClassifier.model() != null ? srcClassifier.model().getClassifier(opname)
```
Clarification added.
